android/sdk/src/main/java/com/taobao/weex/ui/component/WXScroller.java

```diff
@@ -482,44 +485,45 @@ public void onScrollChanged(WXHorizontalScrollView scrollView, int x, int y, int
       scrollView.addView(mRealView, layoutParams);
       scrollView.setHorizontalScrollBarEnabled(false);
       mScrollerView = scrollView;
-        final WXScroller component = this;
-        final View.OnLayoutChangeListener listener = new View.OnLayoutChangeListener() {
-          @Override
-          public void onLayoutChange(View view, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
-            final View frameLayout = view;
-            scrollView.post(new Runnable() {
-              @Override
-              public void run() {
-                if (isLayoutRTL()) {
-                  int mw = frameLayout.getMeasuredWidth();
-                  scrollView.scrollTo(mw, component.getScrollY());
-                } else {
-                  boolean scrollToBegin = true;
-                  Object scrollToBeginOnLTR = getAttrs().get("scrollToBegin");
-                  if (scrollToBeginOnLTR instanceof String){
-                    if("false".equalsIgnoreCase((String)scrollToBeginOnLTR)){
-                      scrollToBegin = false;
-                    }
-                  }
-                  if (scrollToBegin){
-                    scrollView.scrollTo(0, component.getScrollY());
-                  }
+      final WXScroller component = this;
+      final View.OnLayoutChangeListener listener = new View.OnLayoutChangeListener() {
+        @Override
+        public void onLayoutChange(View view, final int left, int top, final int right, int bottom, final int oldLeft, int oldTop, final int oldRight, int oldBottom) {
+          final View frameLayout = view;
+          scrollView.post(new Runnable() {
+            @Override
+            public void run() {
+              if (mIslastDirectionRTL != null && isLayoutRTL() != mIslastDirectionRTL.booleanValue()) {
+                // when layout direction changed we need convert x to RTL x for scroll to the same item
+                int currentX = getScrollX();
+                int totalWidth = getInnerView().getChildAt(0).getWidth();
+                int displayWidth = getInnerView().getMeasuredWidth();
+                scrollView.scrollTo(totalWidth - currentX - displayWidth, component.getScrollY());
+              } else if (isLayoutRTL()) {
+                // if layout direction not changed, but width changede, we need keep RTL offset
+                int oldWidth = oldRight - oldLeft;
+                int width = right - left;
+                int changedWidth = width - oldWidth;
+                if (changedWidth != 0) {
+                  scrollView.scrollBy(changedWidth, component.getScrollY());
```

Comment:
You are missing an `else` case, which could be a problem here.

Reply:
If changedWidth == 0,  we should do nothing